qdev_wrappers/sweep_functions.py

```diff
@@ -30,16 +30,11 @@ def _flush_buffers(*params):
     Supposed to be called inside doNd like so:
     _flush_buffers(inst_set, *inst_meas)
     """
-    instrument_names = set()
-    for param in params:
-        if hasattr(param, '_instrument'):
-            instrument_names.add(param._instrument.root_instrument.name)
-        elif isinstance(param, VisaInstrument):
-            instrument_names.add(param.root_instrument.name)
-
-    for name in instrument_names:
+    for name in set(p.root_instrument.name
```

Comment:
Can we split this in 2 lines, The set comprehension first and the the loop to make it a little bit more readable.